### ISSUE TITLE
5.6.1 segfaults on iPAQ H3650

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4802]
Migrated from [rt.perl.org#8199](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=8199) (status was 'stalled')

Searchable as RT8199$

**Comment by p5pRT** [comment]
### From perl_dummy@bloodgate.com

#### Created by perl_dummy@bloodgate.com



The following program will cause an segfault\, the use integer; seems to be
the cause\, because without it runs\.

If you could tell me how to supply additional information\, I will gladly
fetch them for you\. The machine in question is available trough Compaq's
great TestDrive program\, in case someone want's to try for him/herself\. I
am now trying to compile bleadperl on the machine to see how far that goes
and maybe it cures the problem\.

Thank you in advance\!

\#/usr/bin/perl \-w
&nbsp;  use integer;
&nbsp;  $AND\_BITS = 24;
&nbsp;  $BASE = 100000;
&nbsp;  my $max = 16;
&nbsp;  while \(2 \*\* $max \< $BASE\) \{ $max\+\+; \}

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=critical

Site configuration information for perl v5.6.1:

Configured by buildd at Sun Jun 24 07:28:54 BST 2001.

Summary of my perl5 (revision 5.0 version 6 subversion 1) configuration:
  Platform:
    osname=linux, osvers=2.2.13, archname=arm-linux
    uname='linux medusa 2.2.13 #334 wed dec 29 04:49:54 gmt 1999 armv3l
unknown '
    config_args='-Dccflags=-DDEBIAN -Dcccdlflags=-fPIC -Darchname=arm-linux
-Dprefix=/usr -Dprivlib=/usr/share/perl/5.6.1 -Darc
hlib=/usr/lib/perl/5.6.1 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5
-Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -
Dsitelib=/usr/local/share/perl/5.6.1 -Dsitearch=/usr/local/lib/perl/5.6.1
-Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/ma
n/man3 -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs
-Ud_csh -Uusesfio -Dotherlibdirs=/usr/lib/perl5/5.6:/u
sr/lib/perl5/5.005 -Duseshrplib -Dlibperl=libperl.so.5.6.1 -Dd_dosuid
-Doptimize=-O1 -des'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef
usemultiplicity=undef
    useperlio=undef d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
  Compiler:
    cc='cc', ccflags ='-DDEBIAN -fno-strict-aliasing -I/usr/local/include
-D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O1',
    cppflags='-DDEBIAN -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='2.95.4 20010604 (Debian prerelease)',
gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t',
lseeksize=8
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lgdbm -ldbm -ldb -ldl -lm -lc -lcrypt
    perllibs=-ldl -lm -lc -lcrypt
    libc=/lib/libc-2.2.3.so, so=so, useshrplib=true,
libperl=libperl.so.5.6.1
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.6.1:
    /usr/local/lib/perl/5.6.1
    /usr/local/share/perl/5.6.1
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.6.1
    /usr/share/perl/5.6.1
    /usr/local/lib/site_perl
    /usr/lib/perl5/5.6
    /usr/lib/perl5/5.005
    .


Environment for perl v5.6.1:
    HOME=/house/Tels
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
   
PATH=/house/Tels/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/csh


```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

\-\-\-\-\-BEGIN PGP SIGNED MESSAGE\-\-\-\-\-

Moin\,

On 09\-Jan\-02 perlbugtron@&#8203;bugs\.perl\.org tried to scribble about&#8203;:
> Your e\-mail has been received by the Perl Bug Squashing Team\.

But seems not have been forwarded to p5p for about 12 hours\, so I think it 
got lost somewhere\.
 
>       Original subject&#8203;: 5\.6\.1 segfaults on iPAQ H3650

I can't \(yet\) compile blead on the cute little device\, so I am unsure
whether the core dump is 5\.6\.1 only\, or still present\. Somebody else might
have an idea\.\.\.

The thing is rather a showstopper for BigInt on that device\.

Cheers\,

Tels

> A bug ID \(20020109\.004\) has been assigned and is shown in the subject of
> this email\.  

\- \-\- 
perl \-MMath&#8203;::String \-e 'print \\
Math&#8203;::String\->from\_number\("215960156869840440586892398248"\)\,"\\n"'

 http&#8203;://bloodgate\.com/perl       My current Perl projects
 PGP key available on http&#8203;://bloodgate\.com/tels\.asc or via email\.


\-\-\-\-\-BEGIN PGP SIGNATURE\-\-\-\-\-
Version&#8203;: 2\.6\.3i
Charset&#8203;: latin1

iQEVAwUBPD05yncLPEOTuEwVAQH7gAf9Gimi8OITCMX3H0Zo8TSH0BWp/3/pxe7e
0osut5Q2O\+LPzItB/HT5vnaHA\+P2VpHnto1\+FYD1kAfexHngMY\+lFxKwa786b8OP
iM9rdswajylut0I2l4K\+4Kp/uTPGTeX1ayP6PYO77gsjEuQSAKqA\+luaXs4HEfmo
yBVmlcdp0Zi3POjCuH85gSS508Mz0M3SQJCX6XD0ZEN1ljl5kQWeK1Mpzj19gnE6
zolLqrUVBBRttq8Lw5N\+r1tPVNqRIfpbBu8bvRD8iD\+u9nMgO5ZOO6x7Y05vQTLU
C6HHq8H4dZ52063/80nnQNS5oWEMV2KrX/E3iCszswt0tjzoVPDhfA==
=wxCL
\-\-\-\-\-END PGP SIGNATURE\-\-\-\-\-

**Comment by p5pRT** [comment]
### From @chorny

Is it possible to compile 5\.12\.0 on it?

On Wed Jan 09 03&#8203;:37&#8203;:15 2002\, tels wrote&#8203;:
> This is a bug report for perl from perl\_dummy@&#8203;bloodgate\.com\,
> generated with the help of perlbug 1\.33 running under perl v5\.6\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> The following program will cause an segfault\, the use integer; seems to be
> the cause\, because without it runs\.
> 
> If you could tell me how to supply additional information\, I will gladly
> fetch them for you\. The machine in question is available trough Compaq's
> great TestDrive program\, in case someone want's to try for him/herself\. I
> am now trying to compile bleadperl on the machine to see how far that goes
> and maybe it cures the problem\.
> 
> Thank you in advance\!
> 
> \#/usr/bin/perl \-w
>   use integer;
>   $AND\_BITS = 24;
>   $BASE = 100000;
>   my $max = 16;
>   while \(2 \*\* $max \< $BASE\) \{ $max\+\+; \}
> 
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>     category=core
>     severity=critical
> \-\-\-
> Site configuration information for perl v5\.6\.1&#8203;:
> 
> Configured by buildd at Sun Jun 24 07&#8203;:28&#8203;:54 BST 2001\.
> 
> Summary of my perl5 \(revision 5\.0 version 6 subversion 1\) configuration&#8203;:
>   Platform&#8203;:
>     osname=linux\, osvers=2\.2\.13\, archname=arm\-linux
>     uname='linux medusa 2\.2\.13 \#334 wed dec 29 04&#8203;:49&#8203;:54 gmt 1999 armv3l
> unknown '
>     config\_args='\-Dccflags=\-DDEBIAN \-Dcccdlflags=\-fPIC
\-Darchname=arm\-linux
> \-Dprefix=/usr \-Dprivlib=/usr/share/perl/5\.6\.1 \-Darc
> hlib=/usr/lib/perl/5\.6\.1 \-Dvendorprefix=/usr \-Dvendorlib=/usr/share/perl5
> \-Dvendorarch=/usr/lib/perl5 \-Dsiteprefix=/usr/local \-
> Dsitelib=/usr/local/share/perl/5\.6\.1 \-Dsitearch=/usr/local/lib/perl/5\.6\.1
> \-Dman1dir=/usr/share/man/man1 \-Dman3dir=/usr/share/ma
> n/man3 \-Dman1ext=1 \-Dman3ext=3perl \-Dpager=/usr/bin/sensible\-pager \-Uafs
> \-Ud\_csh \-Uusesfio \-Dotherlibdirs=/usr/lib/perl5/5\.6&#8203;:/u
> sr/lib/perl5/5\.005 \-Duseshrplib \-Dlibperl=libperl\.so\.5\.6\.1 \-Dd\_dosuid
> \-Doptimize=\-O1 \-des'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     usethreads=undef use5005threads=undef useithreads=undef
> usemultiplicity=undef
>     useperlio=undef d\_sfio=undef uselargefiles=define usesocks=undef
>     use64bitint=undef use64bitall=undef uselongdouble=undef
>   Compiler&#8203;:
>     cc='cc'\, ccflags ='\-DDEBIAN \-fno\-strict\-aliasing \-I/usr/local/include
> \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
>     optimize='\-O1'\,
>     cppflags='\-DDEBIAN \-fno\-strict\-aliasing \-I/usr/local/include'
>     ccversion=''\, gccversion='2\.95\.4 20010604 \(Debian prerelease\)'\,
> gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=8
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
> lseeksize=8
>     alignbytes=4\, usemymalloc=n\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='cc'\, ldflags =' \-L/usr/local/lib'
>     libpth=/usr/local/lib /lib /usr/lib
>     libs=\-lgdbm \-ldbm \-ldb \-ldl \-lm \-lc \-lcrypt
>     perllibs=\-ldl \-lm \-lc \-lcrypt
>     libc=/lib/libc\-2\.2\.3\.so\, so=so\, useshrplib=true\,
> libperl=libperl\.so\.5\.6\.1
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-rdynamic'
>     cccdlflags='\-fPIC'\, lddlflags='\-shared \-L/usr/local/lib'
> 
> Locally applied patches&#8203;:
>     
> 
> \-\-\-
> @&#8203;INC for perl v5\.6\.1&#8203;:
>     /usr/local/lib/perl/5\.6\.1
>     /usr/local/share/perl/5\.6\.1
>     /usr/lib/perl5
>     /usr/share/perl5
>     /usr/lib/perl/5\.6\.1
>     /usr/share/perl/5\.6\.1
>     /usr/local/lib/site\_perl
>     /usr/lib/perl5/5\.6
>     /usr/lib/perl5/5\.005
>     \.
> 
> \-\-\-
> Environment for perl v5\.6\.1&#8203;:
>     HOME=/house/Tels
>     LANG \(unset\)
>     LANGUAGE \(unset\)
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
>    
> PATH=/house/Tels/bin&#8203;:/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/bin/X11&#8203;:/usr/games
>     PERL\_BADLANG \(unset\)
>     SHELL=/bin/csh
> 


\-\- 
Alexandr Ciornii\, http&#8203;://chorny\.net

**Comment by p5pRT** [comment]
@dcollinsn - Status changed from 'open' to 'stalled'

**Comment by richardleach** [comment]
Can this ticket be closed?

1. OP hasn't responded since 2002
2. That model of device probably doesn't have a large user pool clamouring for Perl support these days.
3. Supplied test case runs fine on v5.30.0

**Comment by atoomic** [comment]
@dcollinsn I'm closing this case as the discussion is stalled
please let me know if the issue is not fixed in 5.30.0

sincerely
nicolas
